nixos/modules/services/misc/taskserver/default.nix

```diff
@@ -0,0 +1,430 @@
+          Determines whether certain commands are confirmed.
+        '';
+      };
+
+      debug = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Logs debugging information.
+        '';
+      };
+
+      extensions = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Fully qualified path of the Taskserver extension scripts.
+          Currently there are none.
+        '';
+      };
+
+      ipLog = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Logs the IP addresses of incoming requests.
+        '';
+      };
+
+      queueSize = mkOption {
+        type = types.int;
+        default = 10;
+        description = ''
+          Size of the connection backlog, see <citerefentry>
+            <refentrytitle>listen</refentrytitle>
+            <manvolnum>2</manvolnum>
+          </citerefentry>.
+        '';
+      };
+
+      requestLimit = mkOption {
+        type = types.int;
+        default = 1048576;
+        description = ''
+          Size limit of incoming requests, in bytes.
+        '';
+      };
+
+      client = {
+
+        allow = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          example = [ "[Tt]ask [2-9]+" ];
+          description = ''
+            A list of regular expressions that are matched against the reported
+            client id (such as <literal>task 2.3.0</literal>).
+
+            The values <literal>all</literal> or <literal>none</literal> have
+            special meaning. Overidden by any entry in the option
+            <option>services.taskserver.client.deny</option>.
+          '';
+        };
+
+        cert = mkOption {
+          type = types.nullOr types.path;
+          default = null;
+          description = ''
+            Fully qualified path of the client cert. This is used by the
+            <command>client</command> command.
+          '';
+        };
+
+        deny = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          example = [ "[Tt]ask [2-9]+" ];
+          description = ''
+            A list of regular expressions that are matched against the reported
+            client id (such as <literal>task 2.3.0</literal>).
+
+            The values <literal>all</literal> or <literal>none</literal> have
+            special meaning. Any entry here overrides these in
+            <option>services.taskserver.client.allow</option>.
+          '';
+        };
+
+      };
+
+      server = {
+        host = mkOption {
+          type = types.str;
+          default = "localhost";
+          description = ''
+            The address (IPv4, IPv6 or DNS) to listen on.
+          '';
+        };
+
+        port = mkOption {
+          type = types.int;
+          default = 53589;
+          description = ''
+            Port number of the Taskserver.
```

Comment:
The listening host is `localhost` by default, so opening a port wouldn't work here anyway, because access to the loopback interface is unfiltered. However it could make sense add `networking.firewall.allowedTCPPorts` if the host is not `localhost`.